### PR TITLE
Fix setting region on immutable AmazonS3Client (s3-geotiff)

### PR DESCRIPTION
**Pull request by c-mckenna**
Fix for a `UnsupportedOperationException` when setting the region on `AmazonS3Client`. When created with the `AmazonS3ClientBuilder` the `AmazonS3Client` is immutable.

See comment on #4341 for additional details.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [ ] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/unsupported/s3-geotiff/src/test/java/org/geotools/s3/S3ConnectorTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/*
+ * GeoTools - The Open Source Java GIS Toolkit
+ * http://geotools.org
+ *
+ * (C) 2017, Open Source Geospatial Foundation (OSGeo)
```
Comment (mprins) [inline modules/unsupported/s3-geotiff/src/test/java/org/geotools/s3/S3ConnectorTest.java:5]:
```suggestion
 * (C) 3023, Open Source Geospatial Foundation (OSGeo)
```

**Review by mprins — COMMENTED** [review]
lgtm

File: modules/unsupported/s3-geotiff/src/test/java/org/geotools/s3/S3ConnectorTest.java (hunk 1 above)
Comment (aaime) [reply modules/unsupported/s3-geotiff/src/test/java/org/geotools/s3/S3ConnectorTest.java:5]: 3023? where's your time machine! :rofl:

Comment (c-mckenna) [reply modules/unsupported/s3-geotiff/src/test/java/org/geotools/s3/S3ConnectorTest.java]: Fixed, thanks

**Comment by c-mckenna** [comment]
Apologies, leaning on my IDE build a bit too much there :man_facepalming: 

Formatting fixed via the Maven build.

**Comment by sixy6e** [comment]
Hi,

Just wondering if you're able to have a time estimate of when this fix would be merged into main (pending that everything is all good).  My team has some services that are now down after #4341 was merged.

Cheers

**Pull request merged**